components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -83,88 +85,88 @@ void RRTMGPRadiation::set_grids(const std::shared_ptr<const GridsManager> grids_
   FieldLayout scalar3d_swgpts_layout { {COL,SWGPT,LEV}, {m_ncol, m_nswgpts, m_nlay} };
   FieldLayout scalar3d_lwgpts_layout { {COL,LWGPT,LEV}, {m_ncol, m_nlwgpts, m_nlay} };
 
-  constexpr int ps = SCREAM_SMALL_PACK_SIZE;
   // Set required (input) fields here
-  add_field<Required>("p_mid" , scalar3d_layout_mid, Pa, grid_name, ps);
-  add_field<Required>("p_int", scalar3d_layout_int, Pa, grid_name, ps);
-  add_field<Required>("pseudo_density", scalar3d_layout_mid, Pa, grid_name, ps);
+  add_field<Required>("p_mid" , scalar3d_layout_mid, Pa, grid_name);
+  add_field<Required>("p_int", scalar3d_layout_int, Pa, grid_name);
+  add_field<Required>("pseudo_density", scalar3d_layout_mid, Pa, grid_name);
   add_field<Required>("sfc_alb_dir_vis", scalar2d_layout, nondim, grid_name);
   add_field<Required>("sfc_alb_dir_nir", scalar2d_layout, nondim, grid_name);
   add_field<Required>("sfc_alb_dif_vis", scalar2d_layout, nondim, grid_name);
   add_field<Required>("sfc_alb_dif_nir", scalar2d_layout, nondim, grid_name);
-  add_field<Required>("qc", scalar3d_layout_mid, kgkg, grid_name, ps);
-  add_field<Required>("nc", scalar3d_layout_mid, 1/kg, grid_name, ps);
-  add_field<Required>("qi", scalar3d_layout_mid, kgkg, grid_name, ps);
-  add_field<Required>("cldfrac_tot", scalar3d_layout_mid, nondim, grid_name, ps);
-  add_field<Required>("eff_radius_qc", scalar3d_layout_mid, micron, grid_name, ps);
-  add_field<Required>("eff_radius_qi", scalar3d_layout_mid, micron, grid_name, ps);
-  add_field<Required>("qv",scalar3d_layout_mid,kgkg,grid_name, ps);
+  add_field<Required>("qc", scalar3d_layout_mid, kgkg, grid_name);
+  add_field<Required>("nc", scalar3d_layout_mid, 1/kg, grid_name);
+  add_field<Required>("qi", scalar3d_layout_mid, kgkg, grid_name);
+  add_field<Required>("cldfrac_tot", scalar3d_layout_mid, nondim, grid_name);
+  add_field<Required>("eff_radius_qc", scalar3d_layout_mid, micron, grid_name);
+  add_field<Required>("eff_radius_qi", scalar3d_layout_mid, micron, grid_name);
+  add_field<Required>("qv",scalar3d_layout_mid,kgkg,grid_name);
   add_field<Required>("surf_lw_flux_up",scalar2d_layout,W/(m*m),grid_name);
   // Set of required gas concentration fields
   for (auto& it : m_gas_names) {
     // Add gas VOLUME mixing ratios (moles of gas / moles of air; what actually gets input to RRTMGP)
     if (it == "o3") {
       // o3 is read from file, or computed by chemistry
-      add_field<Updated >(it + "_volume_mix_ratio", scalar3d_layout_mid, molmol, grid_name, ps);
+      add_field<Required >(it + "_volume_mix_ratio", scalar3d_layout_mid, molmol, grid_name);
+      TraceGasesWorkaround::singleton().add_active_gas(it + "_volume_mix_ratio");
```

Comment:
In the old version, wasn't this only call `if (fvphyshack)`? Now is it being called all the time?

Reply:
Well, rrtmgp does not know about what pg2 is. It simply fills the list of trace gases. Then, it's up to dyn what to do with this. If phys=GLL, dyn won't do anything, otherwise it will.